saleor/order/calculations.py

```diff
@@ -0,0 +1,281 @@
+from decimal import Decimal
+from typing import Iterable, List, Optional, Tuple
+
+from django.conf import settings
+from django.db.models import prefetch_related_objects
+from django.utils import timezone
+from prices import Money, TaxedMoney
+
+from ..core.prices import quantize_price
+from ..core.taxes import TaxData, TaxError, zero_taxed_money
+from ..plugins.manager import PluginsManager
+from . import ORDER_EDITABLE_STATUS
+from .interface import OrderTaxedPricesData
+from .models import Order, OrderLine
+
+
+def _apply_tax_data_from_manager(
+    manager: PluginsManager, order: Order, lines: Iterable[OrderLine]
+) -> bool:
+    """Apply tax data from manager methods into order and lines.
+
+    If any of the manager order tax methods raises `TaxError`,
+    order and lines stay unchanged.
+
+    :return: True if `TaxError` was raised, False otherwise.
+    """
+    currency = order.currency
+
+    try:
+        price_lines: List[dict] = []
+        for line in lines:
+            variant = line.variant
+            if not variant:
+                continue
+            product = variant.product
+
+            unit = manager.calculate_order_line_unit(order, line, variant, product)
+            total = manager.calculate_order_line_total(order, line, variant, product)
+            tax_rate = manager.get_order_line_tax_rate(
+                order, product, variant, None, unit.undiscounted_price
+            )
+
+            price_lines.append({"tax_rate": tax_rate, "unit": unit, "total": total})
+
+        subtotal = sum((line.total_price for line in lines), zero_taxed_money(currency))
+        shipping_price = manager.calculate_order_shipping(order)
+        shipping_tax_rate = manager.get_order_shipping_tax_rate(order, shipping_price)
+        total = shipping_price + subtotal
+    except TaxError:
+        return True
```

Comment:
I don't understand. Why we are returning `True` when something goes wrong and `False` otherwise? Does it make sense?

Reply:
Can you please try to rewrite this with smaller functions maybe? The current solution is confusing as the function returns unexpected values, according to its name.